src/ja/momonga/src/eu/kanade/tachiyomi/extension/ja/momonga/MomonGA.kt

```diff
@@ -0,0 +1,169 @@
+package eu.kanade.tachiyomi.extension.ja.momonga
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.model.SManga.Companion.COMPLETED
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import keiyoushi.utils.tryParse
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class MomonGA : HttpSource() {
+    override val lang = "ja"
+    override val name = "momon:GA"
+    override val supportsLatest = false
+
+    override val baseUrl = "https://momon-ga.com"
+
+    override val client: OkHttpClient = network.cloudflareClient
+
+    // Chapters
+
+    private val dateFormat = SimpleDateFormat("yyyy年M月d日H時", Locale.ENGLISH)
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val doc = response.asJsoup()
+        return listOf(
+            SChapter.create().apply {
+                name = "単一章"
+                url = response.request.url.encodedPath
+                date_upload = dateFormat.tryParse(doc.select("#post-time").text())
+            },
+        )
+    }
+
+    override fun imageUrlParse(response: Response): String = ""
+
+    // LatestUpdate (Not supported)
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        return MangasPage(emptyList(), false)
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return Request.Builder().build()
+    }
+
+    // Details
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val doc = response.asJsoup()
+
+        var dauthor: String? = null
+        var dartist: String? = null
+        var dgenre: String? = null
+        doc.select("#post-tag > div.post-tag-table").forEach { div ->
+
+            when (div.selectFirst("div.post-tag-title")!!.text()) {
+                "サークル" -> {
+                    dartist = div.select("div.post-tags > a").joinToString(", ") { it.text() }
+                }
+
+                "作者" -> {
+                    dauthor = div.select("div.post-tags > a").joinToString(", ") { it.text() }
+                }
+
+                "内容" -> {
+                    dgenre = div.select("div.post-tags > a").joinToString(", ") { it.text() }
+                }
+
+                else -> {}
+            }
+        }
+
+        return SManga.create().apply {
```

Comment:
Can you move this line to before `doc.select("#post-tag...`? Then you should be able to avoid using `var` variables by setting the values directly.